azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,345 @@
+   }
+
+   @Override
+   public NodeAndInitialCredentials<VMDeployment> createNodeWithGroupEncodedIntoName(
+           final String group, final String name, final Template template) {
+
+      final String location = template.getLocation().getId();
+      DeploymentTemplateBuilder deploymentTemplateBuilder = new DeploymentTemplateBuilder(group, name, template);
+
+      final String loginUser = deploymentTemplateBuilder.getLoginUserUsername();
+      final String loginPassword = deploymentTemplateBuilder.getLoginPassword();
+
+      DeploymentBody deploymentTemplateBody =  deploymentTemplateBuilder.getDeploymentTemplate();
+
+      DeploymentProperties properties = DeploymentProperties.create(deploymentTemplateBody);
+      Gson gson = new GsonBuilder().disableHtmlEscaping().create();
+      org.jclouds.json.Json json = new GsonWrapper(gson);
+
+      final String deploymentTemplate = UrlEscapers.urlFormParameterEscaper().escape(json.toJson(properties));
+
+      logger.debug("Deployment created with name: %s", name);
+
+      final Set<VMDeployment> deployments = Sets.newHashSet();
+      if (!retry(new Predicate<String>() {
+         @Override
+         public boolean apply(final String name) {
+            runningNumber++;
+
+
+            ResourceGroupApi resourceGroupApi = api.getResourceGroupApi();
+            ResourceGroup resourceGroup = resourceGroupApi.get(getGroupId());
+            String resourceGroupName;
+
+            if (resourceGroup == null){
+               final Map<String, String> tags = ImmutableMap.of("description", "jClouds managed VMs");
+               resourceGroupName = resourceGroupApi.create(getGroupId(), location, tags).name();
+            } else {
+               resourceGroupName = resourceGroup.name();
+            }
+
+            Deployment deployment = api.getDeploymentApi(resourceGroupName).create(name, deploymentTemplate);
+
+            if (deployment != null) {
+               VMDeployment vmDeployment = new VMDeployment();
+               vmDeployment.deployment = deployment;
+               deployments.add(vmDeployment);
+            } else {
+               logger.debug("Failed to create deployment!");
+            }
+            return !deployments.isEmpty();
+         }
+      }, azureComputeConstants.operationTimeout(), 1, SECONDS).apply(name)) {
+         final String illegalStateExceptionMessage = format("Deployment %s was not created within %sms so it will be destroyed.",
+                 name, azureComputeConstants.operationTimeout());
+         logger.warn(illegalStateExceptionMessage);
+
+         throw new IllegalStateException(illegalStateExceptionMessage);
+      }
+
+      final VMDeployment deployment = deployments.iterator().next();
+
+
+      return new NodeAndInitialCredentials<VMDeployment>(deployment, name,
+              LoginCredentials.builder().user(loginUser).identity(loginUser).password(loginPassword).authenticateSudo(true).build());
```

Comment:
Also, authenticate sudo should be configured based on the value it has configured in the template options?